include/lbann/layers/io/target/generic_target_layer.hpp

```diff
@@ -44,30 +44,51 @@ class generic_target_layer : public io_layer {
   generic_input_layer *m_paired_input_layer;
   generic_io_buffer *io_buffer;
 
+  /** Ground truth matrix. */
+  AbsDistMat* m_ground_truth;
```

Comment:
Why has the generic _target_layer switched over to using a ground truth matrix rather than staying with the activations matrix.  I realize that the name is a bit clearer, so is the change just for that sake?

Reply:
Target layers don't output to any child layers, so they have no activation matrices. We need to maintain a matrix specifically for the ground truth data.

Reply:
Reconstruction layer is a target layer and could have child layer, I can think of greedy-layerwise autoencoder where reconstruction layers are placed and removed between layers.

Reply:
I think that is more complicated that it needs to be. In my opinion, it would be clearer if we put split layers after the original layer and after the decoder layer and made them the parents of the reconstruction layer. This is also a nicer solution to #195.

Reply:
I think that we should use the split layer to add in the layer-wise reconstruction layer, but I feel like we should talk through it.